packages/protocol/lib/web3-utils.ts

```diff
@@ -387,6 +393,28 @@ export function getFunctionSelectorsForContract(contract: any, contractName: str
   return selectors
 }
 
+export function checkInheritance(baseContractName: string, derivativeContractArtifact: any) {
```

Comment:
Is this function used anywhere? Should it be removed in favor of just the deep version?

Reply:
Personally I think it's worth keeping in case you want to just do a shallow check